config/peerpods/podvm/gcp-podvm-image-handler.sh

```diff
@@ -0,0 +1,346 @@
+    [[ -z "${CAA_REF}" ]] && error_exit "CAA_REF is empty"
+
+    # Ensure booleans are set
+    [[ -z "${INSTALL_PACKAGES}" ]] && error_exit "INSTALL_PACKAGES is empty"
+    [[ -z "${DOWNLOAD_SOURCES}" ]] && error_exit "DOWNLOAD_SOURCES is empty"
+    [[ -z "${CONFIDENTIAL_COMPUTE_ENABLED}" ]] && error_exit "CONFIDENTIAL_COMPUTE_ENABLED is empty"
+    [[ -z "${DISABLE_CLOUD_CONFIG}" ]] && error_exit "DISABLE_CLOUD_CONFIG is empty"
+    [[ -z "${ENABLE_NVIDIA_GPU}" ]] && error_exit "ENABLE_NVIDIA_GPU is empty"
+}
+
+# function to download and install gcloud cli
+
+function install_gcloud_cli() {
+    # Install gcloud cli
+    # If any error occurs, exit the script with an error message
+
+    # Check if gcloud cli is already installed
+    if command -v gcloud &>/dev/null; then
+        echo "gcloud cli is already installed"
+        return
+    fi
+
+    # Download gcloud cli
+		# TODO
+		tee -a /etc/yum.repos.d/google-cloud-sdk.repo << EOM
+[google-cloud-cli]
+name=Google Cloud CLI
+baseurl=https://packages.cloud.google.com/yum/repos/cloud-sdk-el9-x86_64
+enabled=1
+gpgcheck=1
+repo_gpgcheck=0
+gpgkey=https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
+EOM
+
+		dnf install -y google-cloud-cli
+
+    echo "GCP CLI installed successfully"
+}
+
+
+function create_image_using_mkosi() {
+    echo "Creating GCP image using mkosi"
+
+    # If any error occurs, exit the script with an error message
+    # The variables are set before calling the function
+
+		# If PODVM_DISTRO is not set to rhel then exit
+    [[ "${PODVM_DISTRO}" != "rhel" ]] && error_exit "unsupported distro"
+
+    # Set the IMAGE version and name
+    # It should follow the Major(int).Minor(int).Patch(int)
+    GCP_IMAGE_VERSION="${GCP_IMAGE_VERSION_MAJ_MIN}.$(date +'%Y%m%d%S')"
+    GCP_IMAGE_NAME="${GCP_IMAGE_BASE_NAME}-${PODVM_DISTRO}-${GCP_IMAGE_VERSION}"
+
+    cd "${CAA_SRC_DIR}/podvm-mkosi" || error_exit "Failed to change directory to ${CAA_SRC_DIR}/podvm-mkosi"
+
+		# TODO: podvm image customization is made at prepare_source_code method, but
+		# this method is not prepared for mkosi.
+		make binaries
+		make image
+
+		mv build/system.raw build/disk.raw
+		tar -cvzf build/disk.tar.gz -C build disk.raw
+
+		# Create bucket if doesn't exist
+		export GCP_REGION="us-central1"
+		export GCP_BUCKET_NAME="peerpods-bucket"
+		if ! gsutil ls -b "gs://${GCP_BUCKET_NAME}/" &>/dev/null; then
+			gsutil mb -p ${GCP_PROJECT_ID} -l ${GCP_REGION} gs://${GCP_BUCKET_NAME}/
+		fi
+
+		# Upload a disk image to the bucket
+		gsutil cp build/disk.tar.gz gs://${BUCKET_NAME}/${GCP_IMAGE_NAME}.tar.gz
+
+		# Create the image
+		gcloud compute images create ${GCP_IMAGE_NAME} \
+			--source-uri=gs://${BUCKET_NAME}/${GCP_IMAGE_NAME}.tar.gz \
+			--guest-os-features=UEFI_COMPATIBLE
+
+    update_cm_annotation "LATEST_IMAGE_ID" "${GCP_IMAGE_NAME}"
+
+		echo "${GCP_IMAGE_NAME} created successfully"
+}
+
+# Function to create or update podvm-images configmap with all the amis
```

Comment:
To be removed i suppose..

Reply:
I fixed the "amis" part of the comment. However I left the function because I'm not sure if this method is used or not. I tried to find any usage of  `*-podvm-image-handler.sh -R` (for Recreate) but I couldn't find. Do you know in which situation this is called?